### ISSUE TITLE
Training Validation different from Evaluation

### ISSUE THREAD

**Issue by TheSeanLei** [fizyr/keras-retinanet#980]
I'm doing transfer learning with the starting weights given by resnet50_coco_best_v2.1.0.h5. The number of classes is going from 80 to 2 classes and I have a dedicated testing set I use for validation as the model trains.

**Validation during training:**
Epoch 1/50 (going with epoch 1 since it shows the highest mAP)
6624 instances of class person with average precision: 0.6377
0 instances of class vehicle with average precision: 0.0000 (There were vehicles in the training set, and I know I should have vehicles in the testing set but let's focus on the the 'person' precision)
mAP: 0.6377

**Results from evaluate.py on testing set:** converted epoch 1 model to inference model
6624 instances of class person with average precision: 0.0399
0 instances of class vehicle with average precision: 0.0000
mAP: 0.0399

Interestingly enough, the **coco weights** did well during evaluation even though it has 80 classes instead of the 2.
6624 instances of class person with average precision: 0.5917
0 instances of class vehicle with average precision: 0.0000
mAP: 0.5917

Shouldn't the validation and evaluation of the same model produce the same accuracy? Why might the evaluation mAP significantly decrease?

Thanks for your time!

**Comment by hgaiser** [comment]
So if I understand you correctly the validation set during training and the testing set you used for evaluate.py are the same? Yet the mAP is different? What parameters did you use when you trained a network? What were the commands for training, converting and evaluating?

**Comment by hgaiser** [comment]
Most likely fixed by https://github.com/fizyr/keras-retinanet/pull/1042
